drivers/pwm/pwm_stm32.c

```diff
@@ -125,43 +125,135 @@ static int get_tim_clk(const struct stm32_pclken *pclken, uint32_t *tim_clk)
 	} else {
 		apb_psc = CONFIG_CLOCK_STM32_D2PPRE2;
 	}
+#else
+	if (pclken->bus == STM32_CLOCK_BUS_APB1) {
+		apb_psc = CONFIG_CLOCK_STM32_APB1_PRESCALER;
+	}
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) && !defined(CONFIG_SOC_SERIES_STM32G0X)
+	else {
+		apb_psc = CONFIG_CLOCK_STM32_APB2_PRESCALER;
+	}
+#endif
+#endif
 
+#if defined(CONFIG_SOC_STM32F401XC) || defined(CONFIG_SOC_STM32F401XE) ||      \
+	defined(CONFIG_SOC_STM32F411XE) || defined(CONFIG_SOC_STM32F412CG) ||  \
+	defined(CONFIG_SOC_STM32F412ZG) || defined(CONFIG_SOC_STM32F413XX) ||  \
+	defined(CONFIG_SOC_STM32F427XX) || defined(CONFIG_SOC_STM32F437XX) ||  \
+	defined(CONFIG_SOC_STM32F429XX) || defined(CONFIG_SOC_STM32F446XX) ||  \
+	defined(CONFIG_SOC_STM32F469XX) ||                                     \
+	defined(CONFIG_SOC_SERIES_STM32F7X) ||                                 \
+	defined(CONFIG_SOC_SERIES_STM32H7X)
```

Comment:
I wonder whether we could simplify this #if condifion:
Instead of exhaustive list of device,
check CMSIS bitfield definitions:
RCC_DCKCFGR1_TIMPRE   for F7
RCC_DCKCFGR_TIMPRE     for F4
RCC_CFGR_TIMPRE            for H7

Reply:
good point, will also make code ready for socs not present on Zephyr yet.